concent_api/gatekeeper/tests/test_gatekeeper_views.py

```diff
@@ -209,6 +209,159 @@ def test_upload_should_return_401_if_size_of_file_is_wrong(self):
+
+        for value in invalid_values:
+            file = FileTransferToken.FileInfo(
+                path     = 'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+                size     = 1024,
+            )
+
+            self.upload_token.files = [file]
+            self.upload_token.files[0]['checksum'] = value
+
+            golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+            encrypted_token = b64encode(golem_upload_token).decode()
+            response = self.client.post(
+                '{}{}'.format(
+                    reverse('gatekeeper:upload'),
+                    'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend'
+                ),
+                HTTP_AUTHORIZATION             = 'Golem ' + encrypted_token,
+                content_type                   = 'application/x-www-form-urlencoded',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+            )
+
+            self.assertIsInstance(response, JsonResponse)
+            self.assertEqual(response.status_code, 401)
+            self.assertIn('message', response.json().keys())
+            self.assertEqual("application/json", response["Content-Type"])
+
+    @freeze_time("2018-12-30 11:00:00")
+    def test_upload_should_return_401_if_newlines_in_message_path(self):
+        invalid_values = [
+            'blen\nder/benchmark/test_task/scene-Helicopter-2\n7-cycles.blend',
+            '\nblender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend\n',
+        ]
+
+        for value in invalid_values:
+            file = FileTransferToken.FileInfo(
+                path     = value,
+                checksum = 'sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
+                size     = 1024,
+            )
+
+            self.upload_token.files = [file]
+
+            golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+            encrypted_token = b64encode(golem_upload_token).decode()
+            response = self.client.post(
+                '{}{}'.format(
+                    reverse('gatekeeper:upload'),
+                    'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend'
+                ),
+                HTTP_AUTHORIZATION             = 'Golem ' + encrypted_token,
+                content_type                   = 'application/x-www-form-urlencoded',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+            )
+
+            self.assertIsInstance(response, JsonResponse)
+            self.assertEqual(response.status_code, 401)
+            self.assertIn('message', response.json().keys())
+            self.assertEqual("application/json", response["Content-Type"])
+
+    @freeze_time("2018-02-12 16:19:00")
+    def test_upload_should_return_401_if_specific_authorization_header(self):
```

Comment:
What do you mean by 'specific'?

Reply:
It fails on exactly this header. Also if you manipulate a little bit with data which is used to create it, it also fails. But if you create a header from completely different data, it is ok. For example, in test `test_upload_should_return_401_if_newlines_in_message_checksum` it doesn't raised 500 but was handled correctly by gatekeeper validation.

Reply:
Oh, OK. But I think it would have been better to try to describe what's so specific about it in the name of the test. I know it can be hard sometimes but names that are generic like this are often confusing. I'd have to read the whole test carefully to figure out what's going on here.

Reply:
:+1: